### PR TITLE
Extract promotions from the Oscar core.

### PR DESCRIPTION
**Pull request by sasha0**
Refs #2215

### CODE REVIEW

File: src/oscar/apps/dashboard/app.py [hunk 1]
```diff
@@ -54,6 +52,9 @@ def get_urls(self):
             }, name='logout'),
 
         ]
+        if is_promotions_enabled():
+            promotions_app = get_class('dashboard.promotions.app', 'application')
```
Comment (vdboor) [inline src/oscar/apps/dashboard/app.py]: How about `self.promotions_app or get_class(..)` for backwards compatibility with possible overrides?

Comment (sasha0) [reply src/oscar/apps/dashboard/app.py]: I'd probably check this case with the test.

**Comment by solarissmoke** [comment]
This looks OK to me, but a couple of things to consider:

1. It is hard to override the home page view - you have to override the whole `Shop` application and then replace the URL pattern for the home page, which isn't that straightforward. I think a lot of projects will want to replace that view, so we should provide an easier way to do it - not sure what the best approach is but maybe a setting that defines which view class to use.

2. Think we need to have https://github.com/django-oscar/django-oscar-promotions populated before we can merge this?

**Comment by sasha0** [comment]
> It is hard to override the home page view - you have to override the whole Shop application and then replace the URL pattern for the home page, which isn't that straightforward. I think a lot of projects will want to replace that view, so we should provide an easier way to do it - not sure what the best approach is but maybe a setting that defines which view class to use.

How to be with the existing `OSCAR_HOMEPAGE` settings?

> Think we need to have https://github.com/django-oscar/django-oscar-promotions populated before we can merge this?

Indeed.

**Comment by sasha0** [comment]
> It is hard to override the home page view - you have to override the whole Shop application and then replace the URL pattern for the home page, which isn't that straightforward. I think a lot of projects will want to replace that view, so we should provide an easier way to do it - not sure what the best approach is but maybe a setting that defines which view class to use.

After thinking about this for a while, I'm not sure if it not make sense to introduce a new setting, at least for now, since it does not make much difference to have specific `IndexView` or to navigate to the page, intended as a home page.

Anyhow, I didn't have any cases I need to customize `IndexView` on my projects (mostly API or redirection to catalogue or other view through `OSCAR_HOMEPAGE` setting), so I would appreciate any input. If there's no any feedback - I'm inclined just to leave it as is and not introduce new setting.

**Comment by solarissmoke** [comment]
>  I'm inclined just to leave it as is and not introduce new setting.

OK with me.

**Pull request merged**